### PR TITLE
Add polynomial calculation methods for snowfall approximation based on CLASS

### LINKED ISSUE #667: Snow approximation using the CLASS polynomial behaviour
### Description

Implement CLASS calculation method in `snowfall_approximation`

### PR DESCRIPTION
**Pull request by huard**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #667
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (major / minor / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)
- [ ] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Add snowfall estimation methods found in CLASS

### Does this PR introduce a breaking change?

Yes. I changed the condition for the binary method to match CLASS behaviour. Precip is now snow when tas <= 0, instead of tas < 0.

### CODE REVIEW

File: xclim/indices/_conversion.py
Comment (aulemahal) [inline xclim/indices/_conversion.py:612]:
```suggestion
        thresh = convert_units_to(thresh, tas)
        upper = convert_units_to("2 degC", tas) + thresh
```
One less `convert_units_to`  calls!

File: xclim/indices/_conversion.py
Comment (aulemahal) [inline xclim/indices/_conversion.py:635]: I believe `polyval` needs a _coordinate_, so a 1D Dataarray with the same name as its dimension and a coord that is itself.
```suggestion
        t = xr.DataArray(t, dims="tas", name="tas", coords={"tas": t})
```

With this you should be able to pass `xr.polyval(t, coeffs)`

File: xclim/indices/_conversion.py
Comment (aulemahal) [inline xclim/indices/_conversion.py:647]: When computing `fraction` on my side, I got a few values above 1. Is it normal? Did I make a mistake?

**Review by aulemahal — APPROVED** [review]
A few suggestions and questions, but otherwise good to go!

File: xclim/indices/_conversion.py
Comment (huard) [reply xclim/indices/_conversion.py:612]: Hum, non, si tu transformes 2C en Kelvin, tu vas te retrouver avec upper ~ 500 K

File: xclim/indices/_conversion.py
Comment (huard) [reply xclim/indices/_conversion.py:635]: Indeed. works. I had tried earlier but without "name".

File: xclim/indices/_conversion.py
Comment (huard) [reply xclim/indices/_conversion.py:647]: If you're within [0, 6]C, it should be bounded within [0,1].

**Comment by huard** [comment]
Should we add `method` to liquid_precip_ratio, winter_rain_ratio, precip_accumulation ?

**Comment by huard** [comment]
@tlogan2000 FYI, a request dating from last spring from CCCS. Shall we wait for ECCC review before merging ?

**Comment by tlogan2000** [comment]
> @tlogan2000 FYI, a request dating from last spring from CCCS. Shall we wait for ECCC review before merging ?

@huard who at ECCC would want a look? Emilia?  Seems relatively straight forward so I would just merge tbh.  We can have Emilia check the new master an adjust if necessary.

File: xclim/indices/_conversion.py
Comment (aulemahal) [reply xclim/indices/_conversion.py:612]: Woups. C'est bien vrai tout ça, :upside_down_face: .

File: xclim/indices/_conversion.py
Comment (aulemahal) [reply xclim/indices/_conversion.py]: Sorry alors:
```python3
import xarray as xr
import numpy as np
import matplotlib.pyplot as plt

# Create nodes for the snowfall fraction: -inf, thresh, ..., thresh+6, inf [degC]
t = np.concatenate(
    [
        [
            -273.15,
        ],
        np.linspace(0, 6, 100, endpoint=False),
        [6, 1e10],
    ]
)
t = xr.DataArray(t, dims="tas")

# I can't get polyval to work directly with a DataArray, hence the empty values serving no purpose.
tmp = xr.DataArray(np.empty_like(t), dims="tas", coords={"tas": t})

# The polynomial coefficients, valid between thresh and thresh + 6 (defined in CLASS)
coeffs = xr.DataArray(
    [100, 4.6664, -15.038, -1.5089, 2.0399, -0.366, 0.0202],
    dims=("degree",),
    coords={"degree": range(7)},
)

fraction = xr.polyval(tmp.tas, coeffs) / 100
fraction[0] = 1
fraction[-2:] = 0

fraction.sel(tas=slice(-1, 1)).plot()
plt.axhline(1)
```
index

Comment (huard) [reply xclim/indices/_conversion.py]: Ah ouais ! Good catch, there is a line in the fortran code that caps at 1... will add that.

**Pull request merged**
